packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -378,6 +389,22 @@ function createMergedTransformStream(
 
 const CLOSE_TAG = '</body></html>'
 
+/**
+ * For chromium based browsers (Chrome, Edge, etc.) and Safari, icons need to stay under <head>
+ * to be picked up by the browser. Firefox doesn't have this requirement.
+ *
+ * Firefox won't work if we insert those icons into head, it will still pick up default favicon.ico.
+ * Because of this limitation, we just don't insert for firefox and leave the default behavior for it.
+ *
+ */
+const createReinsertIconScript = (nonce: string | undefined) => {
+  return `\
+<script ${typeof nonce === 'string' ? `nonce="${nonce}"` : ''}>\
+!/firefox/i.test(navigator.userAgent) && \
```

Comment:
feel like we can just do this in FF too and not encode the browser test. it works fine in FF too

Reply:
firefox will still show the favicon if we do that. example case: navigate from home page to https://icons-order-test.vercel.app/insert-icon-with-script.html